lib/Sema/SemaBounds.cpp

```diff
@@ -797,6 +818,36 @@ namespace {
            Expr *Base = CreateImplicitCast(QT, CastKind::CK_LValueToRValue, E);
            return ExpandToRange(Base, B);
         }
+        case Expr::UnaryOperatorClass: {
+          UnaryOperator *UO = dyn_cast<UnaryOperator>(E);
```

Comment:
Could you use `cast<>` instead of `dyn_cast<>` followed by a check, if what you do in the check is assert?

Reply:
There's a lot of code that uses `dyn_cast` with a check afterwards. We should move over to `cast`, but let's do that in a separate PR after the deadline?